product/product.go

```diff
@@ -33,10 +32,32 @@ type Config struct {
 }
 
 type Product struct {
+	l        hclog.Logger
 	Name     string
 	Seekers  []*seeker.Seeker
 	Excludes []string
 	Selects  []string
+	Config   Config
+}
+
+// Run runs the seekers
+func (p *Product) Run() map[string]interface{} {
+	p.l.Info("Running seekers for", "product", p.Name)
```

Comment:
This is probably a very small issue, but in theory, I think a person could create a Product without using one of the individual `New` commands, in which case `p.l` would be `nil`, and this would panic. I wonder if it would be worth skipping if `p.l` is nil or else assigning it a `hclog.Default()` if it were nil.

Reply:
That's a good point - it's true that it could panic. By convention the New... functions are there in go when the zero values aren't valid so not using it when they're is one would be a bug. The best place to fallback to hclog.Default would be if a logger weren't supplied in New... . Where this gets weird is that the new functions are per-product rather than using a NewProduct fn. I wouldn't expect this to be an issue except maybe for shortcuts in testing 🤔 

If it were anything other than logging i'd be inclined to guard against the nil, but something about multiple lines for logging feels _off_. Idk that's more an aesthetic concern - i'll keep an eye on it but otherwise will punt in this PR.